modules/repository/create.go

```diff
@@ -194,6 +194,10 @@ func CreateRepository(doer, u *user_model.User, opts CreateRepoOptions) (*repo_m
 		}
 	}
 
+	var licenses []string
+	if len(opts.License) > 0 {
+		licenses = append(licenses, opts.License)
+	}
```

Comment:
The license selected here could be different than what `google/licensecheck` so it would change on first push. I think we should remove this for that reason, so license is always obtained from files only.

Reply:
When user create a new repo, `UpdateRepoLicenses` will not be triggered. 
Not sure why it wouldn't be triggered, so I added these codes.
I will try to figure out why later.

Reply:
In another way, I think we should make sure that these licenses selected are as same as what `google/licensecheck` detects.
Otherwise, user select license A, but the license displayed is B which seems like a bug.

Reply:
I added a auto test to figure out the licenses we provided in `options/license` which can not correctly detected by `google/licensecheck`

Reply:
> When user create a new repo, UpdateRepoLicenses will not be triggered.

Right, I guess a repo create from template does not trigger the file change hooks.

> In another way, I think we should make sure that these licenses selected are as same as what google/licensecheck detects.

I guess we can't satisfy all cases as long as google and we do not use the exact same set of supported licenses. I don't see it as a big issue. Just run the detection.

Reply:
There is another package from google: https://github.com/google/licenseclassifier
It have a license db builder, then we can always pull the latest data of SPDX and build the latest license db in makefile?

Reply:
We have a test to promise `opts.license` is same as the result of detection.

@silverwind 
Does this look good to you?